evaluation/diagramLogs.py

```diff
@@ -0,0 +1,57 @@
+import sys
+from datetime import datetime
+
+
+#Takes two times, of format HH:MM:SS and finds the difference
+def calculateTimeDiff(timeOne, timeTwo):
+    postStripOne = datetime.strptime(timeOne, "%H:%M:%S")
+    postStripTwo = datetime.strptime(timeTwo, "%H:%M:%S")
+    if postStripOne.hour == 23 and postStripTwo.hour > 23:
+        correctionPartOne = datetime(2021, 11, 25, 1)
+        correctionPartTwo = datetime(2021, 11, 25, 13)
+        correct = correctionPartOne - correctionPartTwo
+        postStripOne.__add__(correct)
+        postStripTwo.__add__(correct)
+    timeDifference = postStripTwo - postStripOne
+    return int(timeDifference.total_seconds())
+
+
+#Used in finding timing diagrams
+if __name__ == '__main__':
+    resultsSet = {}
+    resultsLink = []
+    for i in range(1, len(sys.argv), 1):
+        linkLines = []
+        #setFlag = False
+        with open(sys.argv[i], "r") as f:
+            fileList = list(f)
+            #prev = None
+            for line in fileList:
+                if line.find("intkey set") != -1:
+                    txNumber = line[15:-1].split()[4]
+                    resultsSet.update({txNumber: (line[0:8])})
+                    #setFlag = True
+                if line.find("show") != -1:
```

Comment:
If a line has the word "show" in it, it checks to see if there is a key with the same tx number in the map. If yes, add to the results